src/pm/hydra/tools/bootstrap/external/slurm_query_node_list.c

```diff
@@ -22,7 +22,7 @@
 static int *tasks_per_node = NULL;
 static struct HYD_node *global_node_list = NULL;
 
-#if defined(HAVE_LIBSLURM)
+#if defined(HAVE_LIBSLURM) && defined(HAVE_SLURM_SLURM_H)
```

Comment:
Shouldn't the right fix to be in the `configure.ac`?

Reply:
Why configure.ac?  I think each `HAVE_` macro stands for the right configure check.

Reply:
We just need configure to tell us whether we should build with libslurm, right? Then if configure does its job correctly, `HAVE_LIBSLURM` should serve that purpose. The `&&` logic can happen in `configure.ac`

Reply:
`#if defined(HAVE_LIBSLURM) && defined(HAVE_SLURM_SLURM_H)` implies there are code that under only `HAVE_LIBSLURM` or only `HAVE_SLURM_SLURM_H`, is that is the case, then this is OK. Otherwise, this is more complex than necessary.

Reply:
Agreed that we only need check whether both header and lib are available. I changed the configure macro.